pkg/processor/build/runtime/pypy/runtime.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2017 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pypy
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/nuclio/nuclio/pkg/common"
+	"github.com/nuclio/nuclio/pkg/errors"
+	"github.com/nuclio/nuclio/pkg/processor/build/runtime"
+	"github.com/nuclio/nuclio/pkg/version"
+)
+
+const (
+	defaultRuntimeVersion = "2-5.9"
+	defaultBaseImageName  = "jessie"
+)
+
+var (
+	supportedRuntimes = map[string]bool{
+		defaultRuntimeVersion: true,
+	}
+
+	supportedImages = map[string]bool{
+		defaultBaseImageName: true,
+	}
+)
+
+type pypy struct {
+	*runtime.AbstractRuntime
+}
+
+// GetProcessorBaseImageName returns the image name of the default processor base image
```

Comment:
Seems like this should share code w/Python runtime with a slight variation of the the `pypy`. I'd really hate to duplicate logic between pypy and cpython.

In fact I see quite a bit of duplication (e.g. getting handlers). Please make a base "Python" struct and have both structs (cpython/pypy) share common code.

Reply:
The only two things in common I see are `getPythonPath` and `resolveFunctionLogger` which IMO should be `AbstractRuntime`. 

Not sure if it's worth adding a base `Python` struct just for one method, IMO it's better to make `getPythonPath` a function and place it in `pkg/common`

Reply:
I'm referring to the *build* runtime. GetProcessorBaseImageName, getFunctionHandler share the exact same logic, but with different parameters.

Also, pkg/common is only for generic functions (i.e. things relevant for other projects as well). I don't want to pollute it with things shared by two specific modules in a specific project.